src/dataset.js

```diff
@@ -174,6 +197,17 @@ export default class Dataset {
     return page;
   }
 
+  /* applies the filter to the resolvedPage at the given index
+   * Returns the page at the given index */
+  _filterPage(pages, i) {
+    let page = pages[i];
+    if(page && page.isResolved) {
+      page = page.resolve(page.unfilteredData, page.filterCallback);
```

Comment:
Why call page.filter() when we can just resolve the page with out current attributes?